lib/wal/sources/Scene/SceneManager.hpp

```diff
@@ -0,0 +1,43 @@
+//
+// Created by Zoe Roux on 2021-05-14.
+//
+
+
+#pragma once
+
+#include <queue>
+#include "Scene/Scene.hpp"
+
+namespace WAL
+{
+	//! @brief A class to manage scenes
+	class SceneManager
+	{
+	private:
+		std::deque<Scene> _scenes = {};
+	public:
+		//! @brief Add a scene to the container and move to it.
+		//! @return The manager instance used to call this function is returned. This allow method chaining.
+		SceneManager &addScene(Scene &&scene);
+
+		//! @brief Add a scene before the current scene to allow
+		//! @return The manager instance used to call this function is returned. This allow method chaining.
+		SceneManager &addBackScene(Scene &&scene);
```

Comment:
Doc should clearer (I had to read it 3 times to understand)
method should be named addSecondScene (not a great name but a name, but the name should be enough to understand what the function does)
I don't know how it will be useful